### ISSUE TITLE
Contrastive Loss function parameters

### ISSUE THREAD

**Issue by emilysilcock** [UKPLab/sentence-transformers#1567]
Thanks so much for this great library. 

I've been using your contrastive loss function to train for a pairwise similarity task and I was wondering if you could give some insight into the margin parameter on this loss function. I've used contrastive loss functions that haven't had such a parameter, so I'm wondering why one is needed in this case. I guess my rough thought it that shouldn't it be enough to push apart different pairs and pull apart different ones? Why is a threshold necessary for this? 

Thanks!
Emily

**Comment by nreimers** [comment]
Which one do you use exactly?

**Comment by emilysilcock** [comment]
For the moment `sentence_transformers.losses.OnlineContrastiveLoss()` following `https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/quora_duplicate_questions/training_OnlineContrastiveLoss.py` - thanks!

**Comment by nreimers** [comment]
In order that embeddings are not pushed infinitely apart, there is a margin. Once the distance is larger than the margin, the loss will be zero.
